qiskit_ibm_runtime/session.py

```diff
@@ -120,18 +118,36 @@ def __init__(
         self._instance = None
         if isinstance(backend, IBMBackend):
             self._instance = backend._instance
+            sim_backend = backend.simulator
             backend = backend.name
+        else:
+            backend_obj = self._service.backend(backend)
+            self._instance = backend_obj._instance
+            sim_backend = backend_obj.simulator
         self._backend = backend
 
-        self._setup_lock = Lock()
-        self._session_id: Optional[str] = None
+        if not sim_backend:
+            self._session_id = self._create_session(self._backend, self._instance)
+        else:
+            self._session_id = None
+
```

Comment:
We need to add this logic because on IQP, POST `/sessions` requires `backend` and `instance` and will result in an error if a simulator is used